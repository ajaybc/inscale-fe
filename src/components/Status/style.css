.Status__status {
  width: 10px;
  height: 10px;
  display: inline-block;
  margin-right: 5px;
  border-radius: 5px;
  text-transform: capitalize;
}

.Status__status--active {
  background: #1ba94c;
}

.Status__status--inactive {
  background: #da235e;
}