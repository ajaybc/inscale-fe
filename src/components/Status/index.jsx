import React, {Fragment} from 'react';
import moment from 'moment';

import './style.css';

const today = moment();

//TODO: Check status and display color accordingly
export default ({startDate, endDate}) => {
  const status = (startDate < today && today < endDate)?'active':'inactive';
  return <Fragment>
    <div className={`Status__status Status__status--${status}`}></div> {status}
  </Fragment>
} 