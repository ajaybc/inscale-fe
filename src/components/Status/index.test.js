import React from 'react';
import moment from 'moment';
import renderer from 'react-test-renderer';

import Status from './index.jsx';

it('Status displays active when current date is between the start and end date', () => {
  const tree = renderer
    .create(<Status startDate={moment().add(-5, 'days')} endDate={moment().add(5, 'days')}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Status displays inactive when current date is not between the start and end date', () => {
  const tree = renderer
    .create(<Status startDate={moment().add(-5, 'days')} endDate={moment().add(-2, 'days')} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});