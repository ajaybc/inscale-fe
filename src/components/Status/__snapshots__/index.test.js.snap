// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status displays active when current date is between the start and end date 1`] = `
Array [
  <div
    className="Status__status Status__status--active"
/>,
  " ",
  "active",
]
`;

exports[`Status displays inactive when current date is not between the start and end date 1`] = `
Array [
  <div
    className="Status__status Status__status--inactive"
/>,
  " ",
  "inactive",
]
`;
